### DISCUSSION TITLE
Adding ASN, Network, and Organization data to Zeek logs in Elastic

### DISCUSSION THREAD

**Discussion by jollysig** [Security-Onion-Solutions/securityonion#5688] [Unsupported Versions] [answered]
I would like to enrich the GeoIP data in Elastic to also include ASN, Network, and Organization Name. I see that Elastic supports that GeoLite2-ASN database and is already using other GeoLite databases to add the location data.

I added this to the "common" ingest node pipeline and didn't see any difference when looking at the zeek logs in Kibana:

[
  {
    "geoip": {
      "field": "destination.ip",
      "target_field": "destination.geo",
      "database_file": "GeoLite2-City.mmdb",
      "ignore_missing": true,
      "ignore_failure": true,
      "properties": [
        "ip",
        "country_iso_code",
        "country_name",
        "continent_name",
        "region_iso_code",
        "region_name",
        "city_name",
        "timezone",
        "location"
      ]
    }
  },
  {
    "geoip": {
      "field": "destination.ip",
      "target_field": "destination_asn.geo",
      "database_file": "GeoLite2-ASN.mmdb",
      "ignore_missing": true,
      "ignore_failure": true,
      "description": "Adds ASN and Network CIDR based on destination.ip"
    }
  },
  {
    "geoip": {
      "field": "source.ip",
      "target_field": "source.geo",
      "database_file": "GeoLite2-City.mmdb",
      "ignore_missing": true,
      "ignore_failure": true,
      "properties": [
        "ip",
        "country_iso_code",
        "country_name",
        "continent_name",
        "region_iso_code",
        "region_name",
        "city_name",
        "timezone",
        "location"
      ]
    }
  },
  {
    "geoip": {
      "field": "source.ip",
      "target_field": "source_asn.geo",
      "database_file": "GeoLite2-ASN.mmdb",
      "ignore_missing": true,
      "ignore_failure": true,
      "description": "Adds ASN and Network CIDR based on source.ip"
    }
  },

Does anyone know what I am missing to get this to work?

Thanks

**Comment by dougburks** [comment, marked as answer]
We just merged a pull request for ASN and so you should see this in the next release:
https://github.com/Security-Onion-Solutions/securityonion/pull/5069

**Reply by jollysig** [reply]
Thanks Doug!
